### PR TITLE
Make ambiguous commit error a retryable SQL error

### LINKED ISSUE #10247: teamcity: failed tests on master: acceptance/TestNodeRestart
The following tests appear to have failed:

[#37122](https://teamcity.cockroachdb.com/viewLog.html?buildId=37122):

```
--- FAIL: acceptance/TestNodeRestart (12.883s)
	chaos_test.go:270: pq: transaction commit result is ambiguous

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by spencerkimball**
Fixes #10247

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/10248)

### CODE REVIEW

File: pkg/testutils/error.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func IsPError(pErr *roachpb.Error, re string) bool {
 // that show a connection issue or an issue with the node itself. This can
 // occur when a node is restarting or is unstable in some other way.
 func IsSQLRetryError(err error) bool {
-	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF)")
+	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF|commit result is ambiguous)")
```
Comment (petermattis) [inline pkg/testutils/error.go]: Isn't an ambiguous commit error explicitly non-retryable? Seems like this function name is confusing/inaccurate.

Comment (spencerkimball) [inline pkg/testutils/error.go]: It's definitely not explicitly non-retryable. In fact, I'd argue it's mostly something you'd like to retry, as in many cases you'd want to retry it until you get a constraint violation error or success.

Comment (petermattis) [inline pkg/testutils/error.go]: Unless your primary-key contains a `SERIAL` column.

**Review by petermattis — APPROVED** [review]
LGTM

**Comment by tschottdorf** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/testutils/error.go, line 58 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/10248#-KV1qOGDs1KFL60W8J5v-r1-58:-KV1uXhHO2s3pJFxkPwN:b-z1cwyy) ([raw file](https://github.com/cockroachdb/cockroach/blob/6ac54fad38721f01f320283d1ea521e0f343aed4/pkg/testutils/error.go#L58)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Unless your primary-key contains a `SERIAL` column.
</details>
It's a retriable (as in non-definitive) error, but not a retry error. Maybe a rename is enough to solve this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/10248)*

**Comment by spencerkimball** [comment]
OK, will rename. PTAL

File: pkg/testutils/error.go [hunk 2]
```diff
@@ -51,11 +51,12 @@ func IsPError(pErr *roachpb.Error, re string) bool {
...
 
-// IsSQLRetryError returns true if err is retryable. This is true for errors
-// that show a connection issue or an issue with the node itself. This can
-// occur when a node is restarting or is unstable in some other way.
-func IsSQLRetryError(err error) bool {
-	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|EOF)")
+// IsSQLRetryableError returns true if err is retryable. This is true
+// for errors that show a connection issue or an issue with the node
+// itself. This can occur when a node is restarting or is unstable in
+// some other way.
```
Comment (tschottdorf) [inline pkg/testutils/error.go]: Perhaps add
> // The query may have run to completion regardless of this error.

**Review by tbg — APPROVED** [review]

File: pkg/testutils/error.go (hunk 2 above)
Comment (spencerkimball) [inline pkg/testutils/error.go]: OK

**Pull request merged**